arquillian/common/src/main/java/tck/arquillian/protocol/common/TsTestPropsBuilder.java

```diff
@@ -176,6 +176,9 @@ public static String[] runArgs(ProtocolCommonConfig config, Deployment deploymen
                 if(propValue.startsWith("${") && propValue.endsWith("}")) {
                     String refName = propValue.substring(2, propValue.length() - 1);
                     propValue = tsJteProps.getProperty(refName);
+                    if(propValue == null && refName != null) {
+                        propName = System.getProperty(refName);
```

Comment:
Maybe I'm reading this completely wrong, but should this not be propValue = System.getProperty(refName);?

Reply:
I think you are right as the read propValue is in the form of ${property name} and if that property is not found, we should try to read the system property `refName` into propValue.  If that is null, the last fallback could be to read propName from System.properties.